App/Containers/AddCaptionScreen.js

```diff
@@ -1,110 +1,79 @@
 import React from 'react'
 import { connect } from 'react-redux'
-import { View, Text, Image, ScrollView } from 'react-native'
+import { View } from 'react-native'
 import { TextileHeaderButtons, Item } from '../Components/HeaderButtons'
 import Input from '../SB/components/Input'
-import { Button } from 'react-native-elements'
 import { NavigationActions } from 'react-navigation'
 import styles from '../SB/views/ThreadCreate/statics/styles'
 import UIActions from '../Redux/UIRedux'
-import CameraRollActions from '../Redux/CameraRollRedux'
-
 
 class AddCaptionScreen extends React.Component {
-  constructor (props) {
-    super(props)
-    this.state = {
-      value: ''
-    }
-  }
-
   static navigationOptions = ({ navigation }) => {
     const params = navigation.state.params || {}
     return {
       headerTitle: 'Photo Caption',
       headerLeft: (
         <TextileHeaderButtons left>
-          <Item title='Back' iconName='arrow-left' onPress={() => { navigation.dispatch(NavigationActions.back()) }} />
+          <Item title='Back' iconName='arrow-left' onPress={() => {
+            params.cancelShare()
+            navigation.dispatch(NavigationActions.navigate({ routeName: params.backTo }))
+          }} />
         </TextileHeaderButtons>
       ),
       headerRight: params.submitEnabled && (
         <TextileHeaderButtons>
-          <Item title='Save' onPress={() => params.submit()} />
+          <Item title='Share' onPress={() => {
+            params.share()
+            navigation.dispatch(NavigationActions.navigate({ routeName: params.backTo }))
+          }} />
         </TextileHeaderButtons>
       ),
     }
   }
 
   handleNewText = (text: string) => {
-    this.setState({ value: text})
+    this.props.updateComment(text)
     this.props.navigation.setParams({
-      close: () => { this.props.close() },
```

Comment:
This is cool, I discovered that `navigation.setParams` doesn't overwrite any params that were previously set, it just updates or adds to them. So you don't have to re-set the params that were already set in `componentWillMount`